src/util/coordinate_calculation.cpp

```diff
@@ -34,6 +34,13 @@ std::uint64_t squaredEuclideanDistance(const Coordinate lhs, const Coordinate rh
 
 double haversineDistance(const Coordinate coordinate_1, const Coordinate coordinate_2)
```

Comment:
I would keep the original function as `haversineDistance` and the new one rename to `greatCircleDistanceFCC`,
Also `greatCircleDistance` must have the same interface and can be used in  a proxy function like `calculateDistance() { return greatCircleDistanceFCC(); } `.  It is really helpful to have a fast flip between "exact" and approximated distance values.